api/src/test/scala/com/rasterfoundry/granary/PredictionServiceSpec.scala

```diff
@@ -160,4 +234,62 @@ class PredictionServiceSpec
       createdPrediction ==== predictionById
     }
   }
+
+  def addPredictionResultsExpectation = prop {
+    (model: Model.Create, prediction: Prediction.Create) =>
+      {
+        val testIO = for {
+          createdModel <- createModel(model, modelService)
+          createdPrediction <- createPrediction(
+            prediction.copy(modelId = createdModel.id),
+            predictionService
+          )
+          message = if (Random.nextFloat > 0.5) PredictionSuccess("s3://foo/bar.geojson")
+          else PredictionFailure("model failed sorry sorry sorry")
+          update1 <- updatePrediction[Prediction](
+            message,
+            createdPrediction,
+            createdPrediction.webhookId.get
+          )
+          // Do it again, with the same webhook id
+          update2 <- updatePredictionRaw(
+            message,
+            createdPrediction,
+            createdPrediction.webhookId.get
+          )
+          update2Body <- OptionT.liftF { update2.as[NotFound] }
+          // Do it again, but with a random ID for the webhook
+          update3 <- updatePredictionRaw(
+            message,
+            createdPrediction,
+            UUID.randomUUID
+          )
+          update3Body <- OptionT.liftF { update3.as[NotFound] }
+          _           <- deleteModel(createdModel, modelService)
+        } yield { (message, update1, update2, update2Body, update3, update3Body) }
+
+        val (msg, successfulUpdate, conflictUpdateResp, _, noWebhookUpdateResp, _) =
+          testIO.value.unsafeRunSync.get
+
+        val updateExpectation =
+          msg match {
+            case PredictionSuccess(_) =>
+              List(
+                successfulUpdate.outputLocation !=== None,
+                successfulUpdate.status ==== JobStatus.Successful,
+                successfulUpdate.statusReason ==== None
+              )
+            case PredictionFailure(_) =>
+              List(
+                successfulUpdate.outputLocation ==== None,
+                successfulUpdate.status ==== JobStatus.Failed,
+                successfulUpdate.statusReason !=== None
+              )
+          }
+
+        conflictUpdateResp.status.code ==== 404 &&
```

Comment:
This is where the 409 conflict expectation would be, if we wanted to go that route -- it's a testable strategy, but a little messier